### PR TITLE
Change IssueSelectionService not to listen to AnalysisIssueSelectionService event

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Part of #2023

### CODE REVIEW

**Comment by rita-g-sonarsource** [comment]
@duncanp-sonar I couldn't remove the Set from the detailed selection service interface, because the outer service is the one setting it.. https://github.com/SonarSource/sonarlint-visualstudio/blob/branch-taint/src/IssueViz/Selection/IssueSelectionService.cs#L71

I looked into merging the two services into 1, but it becomes a bit messy: I need to store 2 issue vizs, ("SelectedIssue" and "SelectedIssueWithSecondaryLocations") and the code gets a bit hard to follow.
I also tried having 1 class that implements both interfaces but delegates the implementation, e.g.:
```
    [Export(typeof(IIssueSelectionService))]
    [Export(typeof(IAnalysisIssueSelectionService))]
    [PartCreationPolicy(CreationPolicy.Shared)]
    internal sealed class IssueSelectionService : IIssueSelectionService, IAnalysisIssueSelectionService
    {
        private readonly AnalysisIssueSelectionService flowStepSelectionService;
      
        public IssueSelectionService()
        {
            flowStepSelectionService = new AnalysisIssueSelectionService(); // use the real class and call its Set that is now internal and not part of the interface
        }
```
... but that is messy as well. So I'm not sure. WDYT?

**Comment by duncanp-sonar** [comment]
From the point of view of the "primary" selection, isn't the detailed selection service a sink?
i.e. the detailed selection service should be listening to the events from the "primary" selection service and reacting appropriately. 

The primary selection service is more general; it shouldn't need to know that the more specialised detailed selection service exists.

**Review by duncanp-sonar — APPROVED** [review]

**Comment by rita-g-sonarsource** [comment]
> The primary selection service is more general; it shouldn't need to know that the more specialised detailed selection service exists.

Doh! You're right!

**Comment by duncanp-sonar** [comment]
So today's score for obvious design issues spotted by the reviewer is currently 1-1 :-)
